src/basecamp.cpp

```diff
@@ -23,24 +64,155 @@ std::string basecamp::board_name() const
     return string_format( _( "%s Board" ), name.c_str() );
 }
 
-std::string basecamp::save_data() const
+static expansion_data parse_expansion( const std::string &terrain, const tripoint &new_pos )
 {
-    std::ostringstream data;
+    expansion_data e;
+    int last_bar = terrain.find_last_of( '_' );
+    e.type = terrain.substr( 13, last_bar - 13 );
```

Comment:
Is 13 length of "faction_base_"? Maybe move it to static const variable too?

Reply:
Fixed, thanks.